### ISSUE TITLE
How to deal with this library in WebAssembly

### ISSUE THREAD

**Issue by HerrNamenlos123** [sago007/PlatformFolders#28]
Hi there, thank you for this amazing library. I used it everywhere in the past years.
Now, I am porting my desktop application framework to also run in browsers using Emscripten/WebAssembly. 
The idea is to streamline app development and to be able to create apps that out of the box run on desktop, and in the browser.

Now, the library compiles fine in Emscripten, no issues there. However, if I run the program, I get an exception "Unable to get passwd struct". I see this is normal behavior, as it seems to call a system function, that is presumably doomed to fail in a web environment. 
This is to be expected.

Now, you seem to have more knowledge than me about filesystems and file locations in general. Have you ever thought about WebAssembly or how your library would relate to browser environments?
I could just bypass the library and hard-code file locations, but do you know of any standards of file locations in a browser? Or virtual file systems?

TLDR there is nothing wrong with the library but I am wondering if you ever thought about it and how you would handle it.

**Comment by sago007** [comment]
I recently had this problem myself while building for Emscripten and I have added a commit for it https://github.com/sago007/PlatformFolders/commit/5976127493a5dbbc505a668fabddbd4c87d01d53

The problem was that Emscripten runs as root. This triggers an old protection for Ubuntu's sudo implementation (which is fixed today and the protection could be removed).

I recently build a web version of my game Block Attack and I used "FS.filesystems.IDBFS" with the dir:  https://github.com/blockattack/blockattack-game/blob/sago_dev/source/code/os.cpp

I do not have any other good inputs. Emscripten can never share data with other users anyway but this way I have minimal differences between the web and the native build.
